orderbook-subgraph/schema.graphql

```diff
@@ -0,0 +1,99 @@
+type MarketData @entity {
+  "ERC1155 TokenID of the CTF Asset"
+  id: ID!
+  "Condition that the token is linked to"
+  condition: String!
+  "Outcome Index, may not be present if an FPMM is not created"
+  outcomeIndex: BigInt
+}
+
+# Exchange OrderFilled event stored directly
+type OrderFilledEvent @entity {
+  "Transaction hash + Order hash"
+  id: ID!
+  "Transaction hash"
+  transactionHash: Bytes!
+  "Timestamp at which filled occurred"
+  timestamp: BigInt!
+  orderHash: Bytes!
+  "Addresses of the maker and the taker"
+  maker: String!
+  taker: String!
+  "Maker assetId"
+  makerAssetId: String!
+  "Taker assetId"
+  takerAssetId: String!
+  "Maker amount filled"
+  makerAmountFilled: BigInt!
+  "Taker amount filled"
+  takerAmountFilled: BigInt!
+  "Fee paid by the order maker"
+  fee: BigInt!
+}
+
+# Exchange OrdersMatched event stored directly
+type OrdersMatchedEvent @entity {
+  "Transaction Hash"
+  id: ID!
+  "Timestamp at which filled occurred"
+  timestamp: BigInt!
+  "Maker asset Id"
+  makerAssetID: BigInt!
+  "Taker asset Id"
+  takerAssetID: BigInt!
+  "Maker amount filled"
+  makerAmountFilled: BigInt!
+  "Taker amount filled"
+  takerAmountFilled: BigInt!
+}
+
+type Orderbook @entity {
+  "Token Id"
+  id: ID!
+  "Number of trades of any kind against this order book"
+  tradesQuantity: BigInt!
+  "Number of purchases of shares from this order book"
+  buysQuantity: BigInt!
+  "Number of sales of shares to this order book"
+  sellsQuantity: BigInt!
+  "Market volume in terms of the underlying collateral value"
+  collateralVolume: BigInt!
+  "Volume scaled by the number of decimals of collateralToken"
+  scaledCollateralVolume: BigDecimal!
+  "Global volume of share purchases in USDC base units"
+  collateralBuyVolume: BigInt!
+  "Global volume of share purchases in USDC scaled by 10^6"
+  scaledCollateralBuyVolume: BigDecimal!
+  "Global volume of share sales in USDC base units"
+  collateralSellVolume: BigInt!
+  "Global volume of share sales in USDC scaled by 10^6"
+  scaledCollateralSellVolume: BigDecimal!
+}
+
+# Exchange: all trades aggregated
+type OrdersMatchedGlobal @entity {
```

Comment:
does this one get updated every time there is a new trade? I wonder if this could generate a bottleneck as it needs to be updated several times per second sometimes (it shouldn't tbh)

Reply:
Yeah gets updated on every trade. I am considering removing this entity entirely since it is not used anywhere afaik, but it might be useful to someone